tests/chainer_tests/functions_tests/math_tests/test_sum.py

```diff
@@ -11,18 +11,29 @@
 from chainer.testing import condition
 
 
+@testing.parameterize(
+    {'dtype': numpy.float16},
+    {'dtype': numpy.float32},
+    {'dtype': numpy.float64},
+)
```

Comment:
I found that this class only tests the tensor of order three. How about using ``testing.product()`` and writing the following?
```python
@testing.parameterize(*testing.product({
    'shape': [(3, 2, 4), (2, 3, 1, 4)],
    'dtype': [numpy.float16, numpy.float32, numpy.float64],
}))
```